src/GafferUI/AnimationGadget.cpp

```diff
@@ -87,9 +87,12 @@ T timeToFrame( float fps, T time )
 	return time * fps;
 }
 
-float snapTimeToFrame( float fps, float time )
+float snapTimeToFrame( float fps, float time, float threshold=std::numeric_limits<float>::max() )
 {
-	return frameToTime( fps, round( timeToFrame( fps, time ) ) );
+	float frame = timeToFrame( fps, time );
+	float rounded = round( frame );
+
+	return frameToTime( fps, abs( frame - rounded ) > threshold ? frame : rounded );
```

Comment:
Should this be `fabs()` rather than `abs()`?

Reply:
I just read [this](https://stackoverflow.com/questions/21392627/abs-vs-stdabs-what-does-the-reference-say) and now my head hurts, but I think the gist was that using `std::abs` is the way to go...thanks for the update!